src/ripple/core/JobCoro.h

```diff
@@ -0,0 +1,74 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CORE_JOBCORO_H_INCLUDED
+#define RIPPLE_CORE_JOBCORO_H_INCLUDED
+
+#include <ripple/core/JobQueue.h>
+#include <beast/win32_workaround.h>
+#include <boost/coroutine/all.hpp>
+
+namespace ripple {
+
+class JobCoro : public std::enable_shared_from_this<JobCoro>
+{
+private:
+    JobQueue& jq_;
+    JobType type_;
+    std::string name_;
+    std::mutex mutex_;
+    boost::coroutines::asymmetric_coroutine<void>::pull_type coro_;
+    boost::coroutines::asymmetric_coroutine<void>::push_type* yield_;
+
+public:
+    template <class F>
+    JobCoro (JobQueue& jq, JobType type, std::string const& name, F&& f)
+        : jq_(jq)
+        , type_(type)
+        , name_(name)
+        , coro_(
+            [&, fn = std::forward<F>(f)]
+            (boost::coroutines::asymmetric_coroutine<void>::push_type& do_yield)
+            {
+                yield_ = &do_yield;
+                this->yield();
+                fn(shared_from_this());
+            })
+    {
+    }
+
+    void resume()
```

Comment:
This function definition should be outside the class declaration and marked `inline`.

Also, each public member in this class including the constructor could use a javadoc comment explaining its operation.